cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -345,12 +344,16 @@ private Action parseAction(String action) {
 
   private void performUpgrade() throws Exception {
     LOG.info("Upgrading System and User Datasets ...");
+    HBaseAdmin hBaseAdmin = new HBaseAdmin(hConf);
     DatasetUpgrader dsUpgrade = injector.getInstance(DatasetUpgrader.class);
     dsUpgrade.upgrade();
+    hBaseTableUtil.dropTable(hBaseAdmin, dsUpgrade.getDatasetInstanceMDSUpgrader().getOldDatasetInstanceTableId());
```

Comment:
You need ``HBaseTableUtil``, ``HBaseAdmin``, ``Configuration`` objects to drop a table. I feel very strongly that classes such as ``FileMetadataManager`` should not know about these classes at all. 

If the Upgrader classes were subclasses of the ``AbstractUpgrader``, then I could have done by having these objects as protected members in ``AbstractUpgrader``. However, seeing that that is not possible under the current design, I found that this approach was a decent compromise for 2.8.0. 

We need to refactor this code to have the Upgrader classes obey a hierarchy eventually. Thoughts?